src/main/java/com/spencerwi/either/EitherCollectors.java

```diff
@@ -0,0 +1,100 @@
+package com.spencerwi.either;
+
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+/**
+ * These collectors collect a stream of Either<L,R> objects to an
+ * Either<List<L>, List<R>> object.
+ *
+ * The left biased collector will produce a left Either if the stream is empty
+ * or contains at least one left Either objects.
+ *
+ * The right biased collector will produce a right Either if the stream is empty
+ * or contains at least one right Either objects.
+ *
+ * @param <L> the "left side" type.
+ * @param <R> the "right side type.
+ */
+
+public class EitherCollectors<L,R> implements Collector< Either<L,R>, EitherCollectors.EitherAccumulator<L,R>, Either<List<L>, List<R>> > {
+    private final boolean leftBiased;
+
+    /**
+    * Factory method for creating a left biased collector
+    */
+    public static <L,R> Collector<Either<L,R>, ?, Either<List<L>, List<R>>> toLeftBiasedList() {
+        return new EitherCollectors<>(true);
+    }
+
+    /**
+     * Factory method for creating a right biased collector
+     */
```

Comment:
I think it'd be useful to move an explanation of what a "left-biased" or "right-biased" list is here.

In reading through the code, I had a hard time wrapping my head around it -- the most confusing piece here, I think, is that this Collector doesn't actually produce a `List`, but instead produces an `Either<List<L>, List<R>>`

Could you walk me through an example use-case for this collector?

Reply:
I've primarily been using the collectors when processing lists of requests or validating lists of expressions. Here I want all requests or expression to be right before proceeding or reject the entire list. 

Example:

```
Either<List<String>, List<ParsedExpression>> = expressionList.stream()
  .map(expr -> parseExpression(expr))
  .collect(EitherCollectors.toLeftBiased());
```
At this point, if one of the expressions was illegal(return a left-either) we want to reject the entire list and report back the error message for each failed expression.